static/app/views/insights/common/utils/useSamplesDrawer.tsx

```diff
@@ -15,36 +15,42 @@ interface UseSamplesDrawerProps {
   Component: React.ReactNode;
   moduleName: ModuleName;
   requiredParams: [string, ...string[]];
+  onClose?: () => void;
 }
 
 export function useSamplesDrawer({
   Component,
   moduleName,
   requiredParams,
+  onClose = () => undefined,
```

Comment:
this opens up the API quite a bit, as an alternative we could provide a list of query parameters which should be removed.

Reply:
That's okay, in fact I think after this PR lands I'll go back and change then all to `onClose` rather than the default for consistency 👍🏻